File: internal/dtos/response/readList.go
package response

import "time"

type ReadListResponse struct {
	ID          string    `json:"id"`
	UserID      string    `json:"user_id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	BookCount   int64     `json:"book_count"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ReadListBookResponse struct {
	Position int64         `json:"position"`
	Book     *BookResponse `json:"book"`
}

type ReadListNextResponse struct {
	Position int64         `json:"position"`
	Book     *BookResponse `json:"book"`
	HasNext  bool          `json:"has_next"`
}

type CBLUnmatchedEntry struct {
	Series string `json:"series"`
	Number string `json:"number"`
	Reason string `json:"reason"`
}

type ImportCBLResponse struct {
	ReadList  *ReadListResponse   `json:"read_list"`
	Total     int                 `json:"total"`
	Matched   int                 `json:"matched"`
	Unmatched []CBLUnmatchedEntry `json:"unmatched"`
}

File: web/src/types/readList.ts
import type { Book } from "./book";

export interface ReadList {
  id: string;
  user_id: string;
  name: string;
  description: string;
  book_count: number;
  created_at: string;
  updated_at: string;
}

export interface ReadListBook {
  position: number;
  book: Book;
}

export interface ReadListNext {
  position: number;
  book?: Book;
  has_next: boolean;
}

export interface CBLUnmatchedEntry {
  series: string;
  number: string;
  reason: string;
}

export interface ImportCBLResult {
  read_list: ReadList;
  total: number;
  matched: number;
  unmatched: CBLUnmatchedEntry[];
}
